File: 2026-09/src/day_11/leet_242/Solution2.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> cnt;
        for (char c : s) cnt[c]++;
        for (char c : t) cnt[c]--;

        for (auto [_, v] : cnt) {
            if (v) return false;
        }

        return true;
    }
};

File: 2026-09/src/day_11/leet_242/Solution2.java
package day_11.leet_242;

import java.util.*;

class Solution2 {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> cnt = new HashMap<>();
        for (char c : s.toCharArray()) {
            cnt.put(c, cnt.getOrDefault(c, 0) + 1);
        }
        for (char c : t.toCharArray()) {
            cnt.put(c, cnt.getOrDefault(c, 0) - 1);
        }

        for (Map.Entry<Character, Integer> e : cnt.entrySet()) {
            if (e.getValue() != 0) return false;
        }

        return true;
    }
}
